sentry-android-navigation/src/test/java/io/sentry/android/navigation/SentryNavigationListenerTest.kt

```diff
@@ -106,14 +163,181 @@ class SentryNavigationListenerTest {
+
+        sut.onDestinationChanged(fixture.navController, fixture.destination, null)
+
+        verify(fixture.hub).startTransaction(
+            check { assertEquals("/github", it) },
+            any(), any(), anyOrNull(), any<Boolean>()
+        )
+    }
+
+    @Test
+    fun `onDestinationChanged starts tracing with destination id if route is not available`() {
+        val sut = fixture.getSut(toRoute = null, hasViewIdInRes = true)
+
+        sut.onDestinationChanged(fixture.navController, fixture.destination, null)
+
+        verify(fixture.hub).startTransaction(
+            check { assertEquals("/id", it) },
+            any(), any(), anyOrNull(), any<Boolean>()
+        )
+    }
+
+    @Test
+    fun `onDestinationChanged captures arguments as additional data for transaction`() {
+        val sut = fixture.getSut(toRoute = "github/{user_id}?per_page={per_page}")
+
+        sut.onDestinationChanged(
+            fixture.navController,
+            fixture.destination,
+            bundleOf("user_id" to 123, "per_page" to 10)
+        )
+
+        verify(fixture.hub).startTransaction(
+            check { assertEquals("/github", it) },
+            any(), any(), anyOrNull(), any<Boolean>()
+        )
+
+        val capturedArgs = fixture.transaction.data!!["arguments"]
+        require(capturedArgs is Map<*, *>)
+        assertEquals(123, capturedArgs["user_id"])
+        assertEquals(10, capturedArgs["per_page"])
+    }
+
+    @Test
+    fun `onDestinationChanged binds transaction to the Scope`() {
+        val sut = fixture.getSut()
+
+        sut.onDestinationChanged(fixture.navController, fixture.destination, null)
+
+        val captor = argumentCaptor<IWithTransaction>()
+        verify(fixture.scope).withTransaction(captor.capture())
+        captor.firstValue.accept(null)
+        verify(fixture.scope).transaction = fixture.transaction
+    }
+
+    @Test
+    fun `onDestinationChanged does not replace existing transaction on the Scope`() {
+        val sut = fixture.getSut()
+
+        sut.onDestinationChanged(fixture.navController, fixture.destination, null)
+
+        val captor = argumentCaptor<IWithTransaction>()
+        verify(fixture.scope).withTransaction(captor.capture())
+        captor.firstValue.accept(mock())
+        verify(fixture.scope, never()).transaction = fixture.transaction
+    }
+
+    @Test
+    fun `onDestinationChanged finishes previous navigation transaction before starting a new one`() {
+        val sut = fixture.getSut()
+
+        sut.onDestinationChanged(fixture.navController, fixture.destination, null)
+        sut.onDestinationChanged(fixture.navController, fixture.destination, null)
+
+        assertEquals(true, fixture.transaction.isFinished)
+        val captor = argumentCaptor<IWithTransaction>()
+        verify(fixture.scope, times(4)).withTransaction(captor.capture())
+        // 1st time - bind to scope, 2nd time - in SentryTracer when finish, 3rd time - in the nav listener
```

Comment:
what's the 4th one?

Reply:
It's again bind to scope, but for the 2nd transaction since we call `onDestinationChanged` 2 times. I think it's irrelevant for this test though, since we are interested only in the 3rd one (though I agree the test is a bit brittle)